cmd/update/update_scheduler.go

```diff
@@ -0,0 +1,157 @@
+	Marshaler: &runtime.JSONPb{
+		MarshalOptions: protojson.MarshalOptions{
+			EmitUnpopulated: true,
+		},
+	},
+}
+
+// updateSchedulerCmd represents the update command
+var updateSchedulerCmd = &cobra.Command{
+	Use:     "scheduler",
+	Short:   "Updates scheduler",
+	Example: "maestro-cli update scheduler ./scheduler.yaml",
+	Long:    "Uses a file (argument) to update a new scheduler on Maestro.",
+	Args:    validateArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, config, err := common.GetClientAndConfig()
+		if err != nil {
+			return err
+		}
+
+		return NewUpdateScheduler(client, config).run(cmd, args)
+	},
+}
+
+type UpdateScheduler struct {
+	client interfaces.Client
+	config *extensions.Config
+}
+
+func NewUpdateScheduler(client interfaces.Client, config *extensions.Config) *UpdateScheduler {
+	return &UpdateScheduler{
+		client: client,
+		config: config,
+	}
+}
+
+func validateArgs(_ *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("missing arg with scheduler config file path")
+	}
+
+	filePath := args[0]
+	_, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("error while reading file: %w", err)
+	}
+
+	return nil
+}
+
+func (cs *UpdateScheduler) run(_ *cobra.Command, args []string) error {
+
+	logger := common.GetLogger()
+	filePath := args[0]
+	bts, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return fmt.Errorf("error reading scheduler file: %w", err)
+	}
+
+	yamls, err := cs.SplitYAML(bts)
+	if err != nil {
+		return fmt.Errorf("error splitting YAML file into multiple objects: %w", err)
+	}
+
+	for _, yaml_object := range yamls {
+
+		schedulerJsonBytes, err := k8s_yaml.YAMLToJSON(yaml_object)
+		if err != nil {
+			return fmt.Errorf("error parsing YAML to Json: %w", err)
+		}
+
+		var request v1.UpdateSchedulerRequest
+		err = protojson.Unmarshal(schedulerJsonBytes, &request)
+		if err != nil {
+			return fmt.Errorf("error parsing Json to v1.UpdateSchedulerRequest: %w", err)
+		}
+
+		serializedRequest, err := marshler.Marshal(&request)
+		if err != nil {
+			return fmt.Errorf("error parsing request to json: %w", err)
+		}
+
+		logger.Debug("updating scheduler: " + request.Name)
+
+		url := fmt.Sprintf("%s/schedulers", cs.config.ServerURL)
+
+		body, status, err := cs.client.Put(url, string(serializedRequest))
+		if err != nil {
+			return fmt.Errorf("error on Put request: %w", err)
+		}
+		if status != http.StatusOK {
+			return fmt.Errorf("update scheduler response not ok, status: %s, body: %s", http.StatusText(status), string(body))
+		}
+
+		var response v1.UpdateSchedulerResponse
+		err = protojson.Unmarshal(body, &response)
+		if err != nil {
+			return fmt.Errorf("error deserializing update scheduler response, details: %w", err)
+		}
+		logger.Info("Successfully executed update scheduler. Operation id: " + response.OperationId)
+	}
+
+	return nil
+}
+
+func (cs *UpdateScheduler) SplitYAML(resources []byte) ([][]byte, error) {
```

Comment:
@caiordjesus i noticed that this function and the marshaled var are duplicated in our code, and them doesn't actually are "related" to this update command or the create one... WDYT about moving this function and the var to a more generic place (e.g common/common.go or another place)?

Reply:
Great! Did it!